### DISCUSSION TITLE
When will the copilot come back

### DISCUSSION THREAD

**Discussion by PabloLION** [community/community#40026] [Copilot Conversations] [answered]
### Select Topic Area

Product Feedback

### Body

Hi all.
I'm a paid user with annual plan. Today, 1h ago, (2022-nov-24 2300 UTC), the copilot stopped working. As it happened a few times recently, I did a similar "repair" (re-log-in to VSCode, reinstall copilot,  reinstall VSCode,  reboot, switch network, VPN) like said in https://github.com/orgs/community/discussions/36190#discussioncomment-4232313. But it seems the `ping copilot-proxy.githubusercontent.com` is down and it's is not reflected in [GitHub Status](https://www.githubstatus.com/)

My questions are
- Is it possible to for us to have an estimated time about when the service would come back? 
- Is it possible to for us to report any anomaly to <https://www.githubstatus.com/>?
Thanks in advance.

**Comment by senecamanu** [comment]
![image](https://user-images.githubusercontent.com/40152234/203872926-795ff05b-26b7-45fa-972f-3df20ed79a74.png)

**Reply by marco-quintella** [reply]
You definitely have a point bro.

**Reply by senecamanu** [reply]
i'm praying

**Comment by bmarden** [comment]
Commenting to hopefully get some visibility on this. It's broken for me as well, same problems outlined by @PabloLION

**Reply by Baivo** [reply]
`[INFO] [auth] [2022-11-24T23:44:57.714Z] Invalid copilot token: missing token: 403 Forbidden
[ERROR] [default] [2022-11-24T23:44:57.717Z] GitHub Copilot could not connect to server. Extension activation failed: "No access to GitHub Copilot found."`

**Reply by drsmile1001** [reply]
Same problem!

**Reply by ravhirizaldi** [reply]
Yeah same

```
[INFO] [auth] [2022-11-25T02:31:01.668Z] Invalid copilot token: missing token: 403 
[ERROR] [default] [2022-11-25T02:31:01.669Z] GitHub Copilot could not connect to server. Extension activation failed: "Thank you for participating in the GitHub Copilot Technical Preview. Your extended access has ended."

**Reply by geekgao** [reply]
> Yeah same
> 
> ```
> [INFO] [auth] [2022-11-25T02:31:01.668Z] Invalid copilot token: missing token: 403 
> [ERROR] [default] [2022-11-25T02:31:01.669Z] GitHub Copilot could not connect to server. Extension activation failed: "Thank you for participating in the GitHub Copilot Technical Preview. Your extended access has ended."
> ```
Me too.

**Comment by zx3xyy** [comment]
[INFO] [auth] [2022-11-25T02:34:37.734Z] Invalid copilot token: missing token: 403 
[ERROR] [default] [2022-11-25T02:34:37.735Z] GitHub Copilot could not connect to server. Extension activation failed: "No access to GitHub Copilot found."

Happens to me as well.

**Comment by davecheney** [comment]
Hello. I'm sorry you had difficulty accessing Copilot. This was caused by a bad deploy which affected a small percentage of Copilot users. The change has been rolled back and you should be able to use Copilot again.

**Reply by PabloLION** [reply]
Yea, I just checked and it works now.

Could we have a better way to actually "report" instead of only "discus" the issue as paid/unpaid users?

**Reply by RMANOV** [reply]
The bug is still here - and blocks many of us - whit great power!

**Reply by 121click** [reply]
I still have bugs, It says Subscription Required.

**Reply by PabloLION** [reply]
Yea. What I hoped was a more systematical way to report / trace the error too. But seems it's not offered here.

**Comment by PabloLION** [comment]
The service is offline again 😤😤😤😤

**Comment by PabloLION** [comment]
Something we can try / workaround with this chance:
- VSCode [Tab Completion](https://code.visualstudio.com/docs/editor/intellisense#_tab-completion)
- Other options of auto completion like [Tabnine](https://www.tabnine.com/), [Amazon CodeWhisperer](https://aws.amazon.com/codewhisperer/), [Replit Ghostwriter](https://replit.com/site/ghostwriter)

**Reply by PabloLION** [reply]
Something interesting to share:
- Tabnine supports [a lot of languages](https://www.tabnine.com/install)
- CodeWhisperer only supports Python, Java, and JS. (I doubt this because of the absence of TS and [here](https://catalog.us-east-1.prod.workshops.aws/workshops/a33a5d69-1417-4d5f-acc9-ae5c7fba665b/en-US) and [here](https://dev.to/aws-builders/code-faster-with-amazon-codewhisperer-jo4) are my search results)
- Ghostwriter can "Refactor a React component" (Copilot lab can so similar things tho poorly) and
- Ghostwriter writes CSS with acceptable style
- Ghostwriter can convert between ECMA versions (in comparison Copilot can't translate python3 to py2, or Java 11 to Java 8 or Java 17)

About the Price
- Tabnine: 15$/month, 12$(annual plan), for the Black Friday it's on 50% discount.
- CodeWhisperer: still in test phase
- Ghostwriter: $7/mo with some repl things

**Reply by PabloLION** [reply]
Tabnine can auto-import while copilot can't.

**Reply by PabloLION** [reply, marked as answer]
Tabnine is not as good as Copilot. And Copilot is back!
